From 0857204bd3853496a1982ac444419b239c4c834e Mon Sep 17 00:00:00 2001
From: Sven van Haastregt <sven.vanhaastregt@arm.com>
Date: Tue, 14 Jun 2022 11:08:55 +0100
Subject: [PATCH 049/172] Update SPIRV-Headers

Update the SHA-1 of the SPIRV-Headers dependency to 5a121866927.
Regenerate the header files using the newer spirv.hpp.  The generated
.h files had to be tweaked manually due to various internal additions.

(cherry picked from commit 3f5e65d55e451f459259afcfe012f834d4fadf59)
---
 lib/SPIRV/libSPIRV/SPIRVIsValidEnum.h |  6 ++++--
 lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h | 26 +++++++++++++++++++++++++-
 spirv-headers-tag.conf                |  2 +-
 tools/spirv-tool/gen_spirv.bash       |  2 +-
 4 files changed, 31 insertions(+), 5 deletions(-)

diff --git a/lib/SPIRV/libSPIRV/SPIRVIsValidEnum.h b/lib/SPIRV/libSPIRV/SPIRVIsValidEnum.h
index ba1dec13..a10e6244 100644
--- a/lib/SPIRV/libSPIRV/SPIRVIsValidEnum.h
+++ b/lib/SPIRV/libSPIRV/SPIRVIsValidEnum.h
@@ -250,8 +250,8 @@ inline bool isValid(spv::BuiltIn V) {
   case BuiltInLayerPerViewNV:
   case BuiltInMeshViewCountNV:
   case BuiltInMeshViewIndicesNV:
-  case BuiltInBaryCoordNV:
-  case BuiltInBaryCoordNoPerspNV:
+  case BuiltInBaryCoordKHR:
+  case BuiltInBaryCoordNoPerspKHR:
   case BuiltInFragSizeEXT:
   case BuiltInFragInvocationCountEXT:
   case BuiltInLaunchIdKHR:
@@ -267,12 +267,14 @@ inline bool isValid(spv::BuiltIn V) {
   case BuiltInWorldToObjectKHR:
   case BuiltInHitTNV:
   case BuiltInHitKindKHR:
+  case BuiltInCurrentRayTimeNV:
   case BuiltInIncomingRayFlagsKHR:
   case BuiltInRayGeometryIndexKHR:
   case BuiltInWarpsPerSMNV:
   case BuiltInSMCountNV:
   case BuiltInWarpIDNV:
   case BuiltInSMIDNV:
+  case BuiltInCullMaskKHR:
   case internal::BuiltInSubDeviceIDINTEL:
   case internal::BuiltInGlobalHWThreadIDINTEL:
     return true;
diff --git a/lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h b/lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h
index dd10da8e..88071dd7 100644
--- a/lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h
+++ b/lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h
@@ -121,6 +121,7 @@ template <> inline void SPIRVMap<Decoration, std::string>::init() {
   add(DecorationPerPrimitiveNV, "PerPrimitiveNV");
   add(DecorationPerViewNV, "PerViewNV");
   add(DecorationPerTaskNV, "PerTaskNV");
+  add(DecorationPerVertexKHR, "PerVertexKHR");
   add(DecorationPerVertexNV, "PerVertexNV");
   add(DecorationNonUniform, "NonUniform");
   add(DecorationNonUniformEXT, "NonUniformEXT");
@@ -128,6 +129,10 @@ template <> inline void SPIRVMap<Decoration, std::string>::init() {
   add(DecorationRestrictPointerEXT, "RestrictPointerEXT");
   add(DecorationAliasedPointer, "AliasedPointer");
   add(DecorationAliasedPointerEXT, "AliasedPointerEXT");
+  add(DecorationBindlessSamplerNV, "BindlessSamplerNV");
+  add(DecorationBindlessImageNV, "BindlessImageNV");
+  add(DecorationBoundSamplerNV, "BoundSamplerNV");
+  add(DecorationBoundImageNV, "BoundImageNV");
   add(DecorationSIMTCallINTEL, "SIMTCallINTEL");
   add(DecorationReferencedIndirectlyINTEL, "ReferencedIndirectlyINTEL");
   add(DecorationClobberINTEL, "ClobberINTEL");
@@ -162,6 +167,8 @@ template <> inline void SPIRVMap<Decoration, std::string>::init() {
   add(DecorationPrefetchINTEL, "PrefetchINTEL");
   add(DecorationStallEnableINTEL, "StallEnableINTEL");
   add(DecorationFuseLoopsInFunctionINTEL, "FuseLoopsInFunctionINTEL");
+  add(DecorationAliasScopeINTEL, "AliasScopeINTEL");
+  add(DecorationNoAliasINTEL, "NoAliasINTEL");
   add(DecorationBufferLocationINTEL, "BufferLocationINTEL");
   add(DecorationIOPipeStorageINTEL, "IOPipeStorageINTEL");
   add(DecorationFunctionFloatingPointModeINTEL,
@@ -270,7 +277,9 @@ template <> inline void SPIRVMap<BuiltIn, std::string>::init() {
   add(BuiltInLayerPerViewNV, "BuiltInLayerPerViewNV");
   add(BuiltInMeshViewCountNV, "BuiltInMeshViewCountNV");
   add(BuiltInMeshViewIndicesNV, "BuiltInMeshViewIndicesNV");
+  add(BuiltInBaryCoordKHR, "BuiltInBaryCoordKHR");
   add(BuiltInBaryCoordNV, "BuiltInBaryCoordNV");
+  add(BuiltInBaryCoordNoPerspKHR, "BuiltInBaryCoordNoPerspKHR");
   add(BuiltInBaryCoordNoPerspNV, "BuiltInBaryCoordNoPerspNV");
   add(BuiltInFragSizeEXT, "BuiltInFragSizeEXT");
   add(BuiltInFragmentSizeNV, "BuiltInFragmentSizeNV");
@@ -301,6 +310,7 @@ template <> inline void SPIRVMap<BuiltIn, std::string>::init() {
   add(BuiltInHitTNV, "BuiltInHitTNV");
   add(BuiltInHitKindKHR, "BuiltInHitKindKHR");
   add(BuiltInHitKindNV, "BuiltInHitKindNV");
+  add(BuiltInCurrentRayTimeNV, "BuiltInCurrentRayTimeNV");
   add(BuiltInIncomingRayFlagsKHR, "BuiltInIncomingRayFlagsKHR");
   add(BuiltInIncomingRayFlagsNV, "BuiltInIncomingRayFlagsNV");
   add(BuiltInRayGeometryIndexKHR, "BuiltInRayGeometryIndexKHR");
@@ -308,6 +318,7 @@ template <> inline void SPIRVMap<BuiltIn, std::string>::init() {
   add(BuiltInSMCountNV, "BuiltInSMCountNV");
   add(BuiltInWarpIDNV, "BuiltInWarpIDNV");
   add(BuiltInSMIDNV, "BuiltInSMIDNV");
+  add(BuiltInCullMaskKHR, "BuiltInCullMaskKHR");
   add(BuiltInMax, "BuiltInMax");
   add(internal::BuiltInSubDeviceIDINTEL, "BuiltInSubDeviceIDINTEL");
   add(internal::BuiltInGlobalHWThreadIDINTEL, "BuiltInGlobalHWThreadIDINTEL");
@@ -390,6 +401,7 @@ template <> inline void SPIRVMap<Capability, std::string>::init() {
   add(CapabilityGroupNonUniformQuad, "GroupNonUniformQuad");
   add(CapabilityShaderLayer, "ShaderLayer");
   add(CapabilityShaderViewportIndex, "ShaderViewportIndex");
+  add(CapabilityUniformDecoration, "UniformDecoration");
   add(CapabilityFragmentShadingRateKHR, "FragmentShadingRateKHR");
   add(CapabilitySubgroupBallotKHR, "SubgroupBallotKHR");
   add(CapabilityDrawParameters, "DrawParameters");
@@ -444,6 +456,7 @@ template <> inline void SPIRVMap<Capability, std::string>::init() {
   add(CapabilityFragmentFullyCoveredEXT, "FragmentFullyCoveredEXT");
   add(CapabilityMeshShadingNV, "MeshShadingNV");
   add(CapabilityImageFootprintNV, "ImageFootprintNV");
+  add(CapabilityFragmentBarycentricKHR, "FragmentBarycentricKHR");
   add(CapabilityFragmentBarycentricNV, "FragmentBarycentricNV");
   add(CapabilityComputeDerivativeGroupQuadsNV, "ComputeDerivativeGroupQuadsNV");
   add(CapabilityFragmentDensityEXT, "FragmentDensityEXT");
@@ -494,6 +507,7 @@ template <> inline void SPIRVMap<Capability, std::string>::init() {
   add(CapabilityStorageTexelBufferArrayNonUniformIndexingEXT,
       "StorageTexelBufferArrayNonUniformIndexingEXT");
   add(CapabilityRayTracingNV, "RayTracingNV");
+  add(CapabilityRayTracingMotionBlurNV, "RayTracingMotionBlurNV");
   add(CapabilityVulkanMemoryModel, "VulkanMemoryModel");
   add(CapabilityVulkanMemoryModelKHR, "VulkanMemoryModelKHR");
   add(CapabilityVulkanMemoryModelDeviceScope, "VulkanMemoryModelDeviceScope");
@@ -514,7 +528,9 @@ template <> inline void SPIRVMap<Capability, std::string>::init() {
   add(CapabilityShaderSMBuiltinsNV, "ShaderSMBuiltinsNV");
   add(CapabilityFragmentShaderPixelInterlockEXT,
       "FragmentShaderPixelInterlockEXT");
+  add(CapabilityDemoteToHelperInvocation, "DemoteToHelperInvocation");
   add(CapabilityDemoteToHelperInvocationEXT, "DemoteToHelperInvocationEXT");
+  add(CapabilityBindlessTextureNV, "BindlessTextureNV");
   add(CapabilitySubgroupShuffleINTEL, "SubgroupShuffleINTEL");
   add(CapabilitySubgroupBufferBlockIOINTEL, "SubgroupBufferBlockIOINTEL");
   add(CapabilitySubgroupImageBlockIOINTEL, "SubgroupImageBlockIOINTEL");
@@ -553,6 +569,7 @@ template <> inline void SPIRVMap<Capability, std::string>::init() {
   add(CapabilityFPGAMemoryAccessesINTEL, "FPGAMemoryAccessesINTEL");
   add(CapabilityFPGAClusterAttributesINTEL, "FPGAClusterAttributesINTEL");
   add(CapabilityLoopFuseINTEL, "LoopFuseINTEL");
+  add(CapabilityMemoryAccessAliasingINTEL, "MemoryAccessAliasingINTEL");
   add(CapabilityFPGABufferLocationINTEL, "FPGABufferLocationINTEL");
   add(CapabilityArbitraryPrecisionFixedPointINTEL,
       "ArbitraryPrecisionFixedPointINTEL");
@@ -560,18 +577,25 @@ template <> inline void SPIRVMap<Capability, std::string>::init() {
   add(CapabilityIOPipesINTEL, "IOPipeINTEL");
   add(CapabilityBlockingPipesINTEL, "BlockingPipesINTEL");
   add(CapabilityFPGARegINTEL, "FPGARegINTEL");
+  add(CapabilityDotProductInputAll, "DotProductInputAll");
   add(CapabilityDotProductInputAllKHR, "DotProductInputAllKHR");
+  add(CapabilityDotProductInput4x8Bit, "DotProductInput4x8Bit");
   add(CapabilityDotProductInput4x8BitKHR, "DotProductInput4x8BitKHR");
+  add(CapabilityDotProductInput4x8BitPacked, "DotProductInput4x8BitPacked");
   add(CapabilityDotProductInput4x8BitPackedKHR,
       "DotProductInput4x8BitPackedKHR");
+  add(CapabilityDotProduct, "DotProduct");
   add(CapabilityDotProductKHR, "DotProductKHR");
+  add(CapabilityRayCullMaskKHR, "RayCullMaskKHR");
   add(CapabilityBitInstructions, "BitInstructions");
+  add(CapabilityGroupNonUniformRotateKHR, "GroupNonUniformRotateKHR");
   add(CapabilityAtomicFloat32AddEXT, "AtomicFloat32AddEXT");
   add(CapabilityAtomicFloat64AddEXT, "AtomicFloat64AddEXT");
   add(CapabilityLongConstantCompositeINTEL, "LongConstantCompositeINTEL");
+  add(CapabilityOptNoneINTEL, "OptNoneINTEL");
   add(CapabilityAtomicFloat16AddEXT, "AtomicFloat16AddEXT");
   add(CapabilityDebugInfoModuleINTEL, "DebugInfoModuleINTEL");
-  add(CapabilityMemoryAccessAliasingINTEL, "MemoryAccessAliasingINTEL");
+  add(CapabilitySplitBarrierINTEL, "SplitBarrierINTEL");
   add(CapabilityGroupUniformArithmeticKHR, "GroupUniformArithmeticKHR");
 
   // From spirv_internal.hpp
diff --git a/spirv-headers-tag.conf b/spirv-headers-tag.conf
index 4388a8e7..9677763a 100644
--- a/spirv-headers-tag.conf
+++ b/spirv-headers-tag.conf
@@ -1 +1 @@
-4995a2f2723c401eb0ea3e10c81298906bf1422b
+5a121866927a16ab9d49bed4788b532c7fcea766
diff --git a/tools/spirv-tool/gen_spirv.bash b/tools/spirv-tool/gen_spirv.bash
index 20ab7e61..61ff2a10 100644
--- a/tools/spirv-tool/gen_spirv.bash
+++ b/tools/spirv-tool/gen_spirv.bash
@@ -163,7 +163,7 @@ genFile() {
 #define ${includeGuard}
 
 #include \"SPIRVEnum.h\"
-#include \"spirv.hpp\"
+#include \"spirv/unified1/spirv.hpp\"
 #include \"spirv_internal.hpp\"
 
 using namespace spv;
-- 
2.20.1

